routes/api/users.js

```diff
@@ -93,4 +103,31 @@ router.get("/current", auth, async (req, res, next) => {
   });
 });
 
+router.patch(
+  "/avatars",
+  auth,
+  upload.single("avatar"),
+  async (req, res, next) => {
+    const { path: tempName, originalname } = req.file;
+    const avatarURL = path.join(AVATAR_DIR, originalname);
+    const { email } = req.user;
+    Jimp.read(tempName)
+      .then((avatar) => {
+        return avatar.resize(250, 250).write(AVATAR_DIR);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+    try {
+      await fs.rename(tempName, avatarURL);
+      await updateAvatar(email, shortAvatarURL(avatarURL));
+    } catch (error) {
+      await fs.unlink(tempName);
+      next(error);
+      return res.status(500).json({ message: "Server error" });
```

Comment:
Albo robisz next(error) albo res.status(), jak będą 2 to wyskoczy error że próbujesz odnieść się do przekazanego już obiektu.

Reply:
Znowu niedopatrzenie, dodałem res.status a zapomniałem usunąć next(error)